test/integration/smoke/test_outofbandmanagement.py

```diff
@@ -0,0 +1,561 @@
+            self.fail("Expected an exception to be thrown, failing")
+        except Exception: pass
+
+        try:
+            cmd = enableOutOfBandManagementForCluster.enableOutOfBandManagementForClusterCmd()
+            response = self.apiclient.enableOutOfBandManagementForCluster(cmd)
+            self.fail("Expected an exception to be thrown, failing")
+        except Exception: pass
+
+        try:
+            cmd = enableOutOfBandManagementForZone.enableOutOfBandManagementForZoneCmd()
+            response = self.apiclient.enableOutOfBandManagementForZone(cmd)
+            self.fail("Expected an exception to be thrown, failing")
+        except Exception: pass
+
+
+    @attr(tags=["smoke", "advanced"])
+    def test_oobm_disable_feature_invalid(self):
+        """
+            Tests out-of-band management host disable feature with
+            invalid options
+        """
+        cmd = self.getOobmDisableCmd()
+        cmd.hostid = -1
+        try:
+            response = self.apiclient.disableOutOfBandManagementForHost(cmd)
+            self.fail("Expected an exception to be thrown, failing")
+        except Exception: pass
+
+        try:
+            cmd = disableOutOfBandManagementForCluster.disableOutOfBandManagementForClusterCmd()
+            response = self.apiclient.disableOutOfBandManagementForCluster(cmd)
+            self.fail("Expected an exception to be thrown, failing")
+        except Exception: pass
+
+        try:
+            cmd = disableOutOfBandManagementForZone.disableOutOfBandManagementForZoneCmd()
+            response = self.apiclient.disableOutOfBandManagementForZone(cmd)
+            self.fail("Expected an exception to be thrown, failing")
+        except Exception: pass
+
+
+    @attr(tags=["smoke", "advanced"])
+    def test_oobm_enable_feature_valid(self):
+        """
+            Tests out-of-band management host enable feature with
+            valid options
+        """
+        self.apiclient.configureOutOfBandManagement(self.getOobmConfigCmd())
+        cmd = self.getOobmEnableCmd()
+        response = self.apiclient.enableOutOfBandManagementForHost(cmd)
+        self.assertEqual(response.hostid, cmd.hostid)
+        self.assertEqual(response.enabled, True)
+
+
+    @attr(tags=["smoke", "advanced"])
+    def test_oobm_disable_feature_valid(self):
+        """
+            Tests out-of-band management host disable feature with
+            valid options
+        """
+
+        self.apiclient.configureOutOfBandManagement(self.getOobmConfigCmd())
+        cmd = self.getOobmDisableCmd()
+        response = self.apiclient.disableOutOfBandManagementForHost(cmd)
+        self.assertEqual(response.hostid, cmd.hostid)
+        self.assertEqual(response.enabled, False)
+
+        response = self.getHost(hostId=cmd.hostid).outofbandmanagement
+        self.assertEqual(response.powerstate, 'Disabled')
+
+
+    @attr(tags=["smoke", "advanced"])
+    def test_oobm_enabledisable_across_clusterzones(self):
```

Comment:
Could this method be separated for zone, cluster, and host?  It seems like it would make the test case more readable.